src/WebJobs.Script.WebHost/WebHostResolver.cs

```diff
@@ -31,14 +34,36 @@ public sealed class WebHostResolver : IDisposable
         private static ScriptSettingsManager _settingsManager;
 
         public WebHostResolver(ScriptSettingsManager settingsManager, ISecretManagerFactory secretManagerFactory,
-            IScriptEventManager eventManager, WebHostSettings settings, IWebJobsRouter router, ILoggerFactoryBuilder loggerFactoryBuilder)
+            IScriptEventManager eventManager, WebHostSettings settings, IWebJobsRouter router, ILoggerProviderFactory loggerProviderFactory,
+            ILoggerProvider fallbackLoggerProvider)
         {
             _settingsManager = settingsManager;
             _secretManagerFactory = secretManagerFactory;
             _eventManager = eventManager;
             _router = router;
-            _loggerFactoryBuilder = loggerFactoryBuilder;
             _settings = settings;
+
+            // loggerProviderFactory is used for creating the LoggerFactory for each ScriptHost
+            _loggerProviderFactory = loggerProviderFactory;
+
+            // fallbackLoggerFactory is used if there is no initialized ScriptHost
```

Comment:
This fallback logger only logs to system logs? Then that would mean you'd have the same issue with lost logs on cold start that I fixed in https://github.com/Azure/azure-webjobs-sdk-script/pull/2228.

The AdminController (and other controllers) are getting this fallback logger. I think you also need to add a default file logger to this as I did in my fix?

Reply:
I wanted to make sure that they got the system logs (for us to debug). But I saw your changes and can add the HostFileLogger as well to this factory.

Question -- how does this work with the standby logger? I'm assuming it has a file system, but where do those files go if it's not specialized? In my current design, that's the logger factory that the standby logger uses as well.

Reply:
... although I can just make sure the standby logger only uses a SystemLogger (without file logging) if it's an issue. It's just one more logging pipeline that you need to reason through when trying to figure out where the logs go.

Reply:
When in standby mode, log directory is in temp. See https://github.com/Azure/azure-webjobs-sdk-script/blob/b0cf84d7e27ac9a37d183a966e9ef7369f50914a/src/WebJobs.Script.WebHost/WebHostResolver.cs#L166